drivers/gpio/gpio_npm1300.c

```diff
@@ -0,0 +1,302 @@
+		}
+	}
+
+	return ret;
+}
+
+static int gpio_npm1300_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	int ret;
+
+	for (size_t idx = 0; idx < NPM1300_GPIO_PINS; idx++) {
+		if ((pins & (1U << idx)) != 0U) {
+			ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, drive[idx]),
+					0U);
+		}
+		if (ret != 0U) {
+			return ret;
+		}
+	}
+
+	for (size_t idx = 0; idx < NPM1300_GPIO_LED_PINS; idx++) {
+		if ((pins & (1U << (idx + NPM1300_GPIO_PINS))) != 0U) {
+			ret = gpio_npm1300_led_set(dev, idx, false);
+		}
+		if (ret != 0U) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static inline int gpio_npm1300_led_configure(const struct device *dev, gpio_pin_t pin,
+					     gpio_flags_t flags)
+{
+	const struct gpio_npm1300_config *config = dev->config;
+	int ret = 0;
+
+	if (flags & GPIO_OUTPUT) {
+		ret = reg_write(dev, NPM_LED_BASE, offsetof(struct led_reg_t, mode[pin]),
+				NPM1300_GPIO_LED_HOST);
+	}
+
+	if (ret != 0U) {
+		return ret;
+	}
+
+	return gpio_npm1300_led_set(dev, pin, (flags & GPIO_OUTPUT_INIT_HIGH) != 0U);
+}
+
+static inline int gpio_npm1300_configure(const struct device *dev, gpio_pin_t pin,
+					 gpio_flags_t flags)
+{
+	const struct gpio_npm1300_config *config = dev->config;
+	int ret = 0;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	if (pin > (NPM1300_GPIO_PINS + NPM1300_GPIO_LED_PINS)) {
+		return -EINVAL;
+	}
+
+	if (pin > NPM1300_GPIO_PINS) {
+		return gpio_npm1300_led_configure(dev, pin - NPM1300_GPIO_PINS, flags);
+	}
+
+	/* Configure mode */
+	if ((flags & GPIO_INPUT) != 0U) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, mode[pin]),
+				NPM1300_GPIO_GPIINPUT);
+	} else if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0U) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, mode[pin]),
+				NPM1300_GPIO_GPOLOGIC1);
+	} else if ((flags & GPIO_OUTPUT) != 0U) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, mode[pin]),
+				NPM1300_GPIO_GPOLOGIC0);
+	}
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Configure open drain */
+	if ((flags & GPIO_SINGLE_ENDED) != 0U) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, opendrain[pin]),
+				1U);
+	} else {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, opendrain[pin]),
+				0U);
+	}
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Configure pulls */
+	if (flags & GPIO_PULL_UP) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, pullup[pin]), 1U);
+	} else {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, pullup[pin]), 0U);
+	}
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (flags & GPIO_PULL_DOWN) {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, pulldown[pin]), 1U);
+	} else {
+		ret = reg_write(dev, NPM_GPIO_BASE, offsetof(struct gpio_reg_t, pulldown[pin]), 0U);
+	}
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Configure drive strength and debounce */
+	if (flags & NPM1300_GPIO_DRIVE_HIGH) {
```

Comment:
I can't access the docs now (infocenter is down?), but, do we call drive high/low or 1ma/6ma?

Reply:
Yes it is 1mA/6mA